src/lib.rs

```diff
@@ -2201,15 +2352,77 @@ impl LightningNode {
             .refund_tx_id)
     }
 
+    pub fn swap_failed_swap_funds_to_lightning(
+        &self,
+        failed_swap_info: FailedSwapInfo,
+        sat_per_vbyte: u32,
+        lsp_fee_param: Option<OpeningFeeParams>,
+    ) -> Result<String> {
+        let swap_address_info = self
+            .generate_swap_address(lsp_fee_param.clone())
+            .map_to_runtime_error(
+                RuntimeErrorCode::NodeUnavailable,
+                "Couldn't generate swap address",
+            )?;
+
+        let prepare_response = self
+            .rt
+            .handle()
+            .block_on(self.sdk.prepare_refund(PrepareRefundRequest {
+                swap_address: failed_swap_info.address.clone(),
+                to_address: swap_address_info.address.clone(),
+                sat_per_vbyte,
+            }))
+            .map_to_runtime_error(RuntimeErrorCode::NodeUnavailable, "Coudln't prepare refund")?;
+
+        let send_amount_sats = failed_swap_info.amount.sats - prepare_response.refund_tx_fee_sat;
+
+        if swap_address_info.min_deposit.sats > send_amount_sats {
+            runtime_error!(
+                RuntimeErrorCode::NodeUnavailable,
```

Comment:
Why that error code? Also for 2 following statements

Reply:
I've added some todo comments. The idea is to use the SDK specific error when there is one. Basically get an implementation very similar to `swap_channel_close_funds_to_lightning`.